asciidoc-confluence-publisher-client/src/main/java/org/sahli/asciidoc/confluence/publisher/client/ConfluencePublisher.java

```diff
@@ -110,32 +135,34 @@ private void deleteConfluenceAttachmentsNotPresentUnderPage(String contentId, Ma
         confluenceAttachmentsToDelete.forEach(this.confluenceClient::deleteAttachment);
     }
 
-
-    private String addOrUpdatePage(String spaceKey, String ancestorId, ConfluencePageMetadata page, String content) {
+    private String addOrUpdatePageUnderAncestor(String spaceKey, String ancestorId, ConfluencePageMetadata page) {
         String contentId;
 
         try {
             contentId = this.confluenceClient.getPageByTitle(spaceKey, page.getTitle());
-            ConfluencePage existingPage = this.confluenceClient.getPageWithContentAndVersionById(contentId);
-            String existingContentHash = this.confluenceClient.getPropertyByKey(contentId, CONTENT_HASH_PROPERTY_KEY);
-            String newContentHash = contentHash(content);
-
-            if (notSameContentHash(existingContentHash, newContentHash)) {
-                this.confluenceClient.deletePropertyByKey(contentId, CONTENT_HASH_PROPERTY_KEY);
```

Comment:
I remove the entire concept of storing the hash as property on the remote page. It is completely unncessary since you have to content to calculate your hash on. It also makes it more performant this way.

Reply:
@cstettler never mind, I know realize why this was made like this. Confluence changes the content after the POST so the content will never be identical to the local content. I will revert it.